packages/azure/data_stream/activitylogs/manifest.yml

```
@@ -41,3 +41,20 @@ streams:
         show_user: false
         description: >-
           Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
+
+      - name: sanitize_newlines
+        type: bool
+        title: Sanitizes New Lines
+        description: Sanitizes New Lines
```

Something like this would probably make more sense. Also, the user might need more context on why these options should be enabled, maybe in the docs?

If I remember correctly no errors would show up in kibana if logs are malformed, how would users know when they need to enable these options? @zmoog 

```suggestion
        description: Removes new lines in logs to ensure proper formatting of JSON data and avoid parsing issues during processing.
```

---

Yep, I would love to have a section explaining this change's context. 

Here's a proposal.

Add a new "malformed JSON" (this name is a placeholder) section in the [general Azure Logs documentation](https://github.com/elastic/integrations/blob/main/packages/azure/_dev/build/docs/README.md); here, we can share that some Azure services may send formally malformed JSON documents. We have a public resource where a Microsoft person states this is happening https://learn.microsoft.com/en-us/answers/questions/1001797/invalid-json-logs-produced-for-function-apps for the Azure Functions logs.

In this section, we can explain how to recognize this scenario. The possible symptoms are:

1. the presence of a `records` array in the `message` field because the input failed to unmarshal the byte slice 
2. There is an `error.message` field with the "Received invalid json from the Azure Cloud platform. Unable to parse the source log message" text.

Side note: after a quick check, it seems the symptom n2 only happens on platform and spring cloud logs, we should probably consider extending the check and `error.message` to other data streams, WDYT?

After giving users more context about this issue, we can share how to address the problem by enabling one of the existing sanitization options available in the advanced settings section.